File: Array/C++/chocolateDistribution.cpp
#include <bits/stdc++.h>
#define log(x) cout<<x<<endl;
using namespace std;

int main()
{
    int m = 5;
    int arr[] = {3, 4, 1, 9, 56, 7, 9, 12};
    int n = sizeof(arr)/sizeof(arr[0]);
    sort(arr, arr+n);
    int min = INT_MAX;
    for(int i=0; i+m-1<n; i++){
        int diff = arr[i+m-1] - arr[i];
        if(diff< min) min = diff;
    }
    log(min)
}

File: Array/Python/chocolateDistribution.py
arr = [3, 4, 1, 9, 56, 7, 9, 12]
m= 5
arr.sort()
n = len(arr)
min_diff = arr[n-1] - arr[0]
for i in range(n-m+1):
    min_diff = min(min_diff, arr[i+m-1]- arr[i])

print(min_diff)
